typed/reconcile_schema.go

```diff
@@ -274,3 +274,18 @@ func typeRefAtPath(t *schema.Map, pe fieldpath.PathElement) (schema.TypeRef, boo
 	}
 	return tr, tr != schema.TypeRef{}
 }
+
+// isUntypedDeducedMap returns true if m has no fields defined, but allows untyped elements.
+// This is equivalent to a openAPI object that has x-kubernetes-preserve-unknown-fields=true
+// but does not have any properties defined on the object.
+func isUntypedDeducedMap(m *schema.Map) bool {
+	return isUntypedDeducedRef(m.ElementType) && m.Fields == nil
+}
+
+func isUntypedDeducedRef(t schema.TypeRef) bool {
+	if t.NamedType != nil {
+		return *t.NamedType == "__untyped_deduced_"
```

Comment:
There are two underscores at the beginning but only one at the end. Is that intentional?

Reply:
Yes, this is now it is defined.

xref: https://github.com/kubernetes/kube-openapi/blob/3c818078ee3de6569a8f02b6345ea3c4cc8b0998/pkg/schemaconv/smd.go#L138, https://github.com/kubernetes-sigs/structured-merge-diff/blob/ea1021dbc0f242313159d5dd4801ff29304712fe/typed/parser.go#L130

Reply:
*eye twitch*